magma/lib/magma/flag.rb

```diff
@@ -1,5 +1,18 @@
 class Magma
   class Flag < Sequel::Model
+    # update_or_create is natively implemented in a later version of Sequel - I believe it can also be installed as a plugin
+    def self.update_or_create(project_name, flag_list)
```

Comment:
This interface is not what i'd expect for update_or_create - in particular the flag_list probably belongs back in the #set controller so the format of params[:flags] is evident in that controller; you might reduce this to lines 7-12 and just take a single hash argument, which would be the standard interface for #update_or_create i think.